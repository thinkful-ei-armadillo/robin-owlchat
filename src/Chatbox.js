import React from 'react';

export default function Chatbox(props){
 const messageType = props.type === 'message' ? 'view': 'hidden';
 const typeOtherThanMessage = props.type === 'message' ? 'hidden': 'view';
 const type = props.type === 'message' ? '' : props.type;
 
 return (
 <div className="chat">
 <div className="messages">
     <span className="chat-time-indicator" data-type-timestamp={props.timestamp}></span>
     <div className={messageType}>
       <p>{props.message}</p>
     </div>
     <div className={typeOtherThanMessage}>
         <strong>{props.name}</strong>
         "gave a {type}"
         <span name={props.type} className={props.type}></span>
     </div>
 </div>
</div>
)
}




