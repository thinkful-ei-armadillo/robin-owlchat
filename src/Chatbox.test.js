import React from 'react';
import ReactDOM from 'react-dom';
import Chatbox from './Chatbox';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Chatbox />, div);
  ReactDOM.unmountComponentAtNode(div);
});
it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Chatbox 
          type= 'message' 
          message='hi'
          timestamp={123344}
          messagetype = "text" />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });