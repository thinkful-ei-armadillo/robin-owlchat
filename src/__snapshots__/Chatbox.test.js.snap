// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="chat"
>
  <div
    className="messages"
  >
    <span
      className="chat-time-indicator"
      data-type-timestamp={123344}
    />
    <div
      className="view"
    >
      <p>
        hi
      </p>
    </div>
    <div
      className="hidden"
    >
      <strong />
      "gave a 
      
      "
      <span
        className="message"
        name="message"
      />
    </div>
  </div>
</div>
`;
